tests/job.go

```diff
@@ -140,7 +140,7 @@ func NewHelloWorldJobUDP(host string, port string) *batchv1.Job {
 // This pod tries to contact the host on the provided port, over HTTP.
 // On success - it expects to receive "Hello World!".
 func NewHelloWorldJobHTTP(host string, port string) *batchv1.Job {
-	check := []string{fmt.Sprintf(`set -x; x="$(head -n 1 < <(curl %s:%s))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, FormatIPForURL(host), port)}
+	check := []string{fmt.Sprintf(`set -x; ping -c 1 %s; x="$(head -n 1 < <(curl %s:%s))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, host, FormatIPForURL(host), port)}
```

Comment:
You wrote in the commit message the problem is just for `nc`? Do we have the same issue with curl?

Reply:
The problem does not reproduce if you open a shell in the job and nc / curl manually. That was what lead me into using this silly work-around.